bolt-modules/boltlib/lib/puppet/functions/apply_prep.rb

```diff
@@ -112,6 +113,13 @@ def apply_prep(target_spec, options = {})
             # Give plan function options precedence over inventory options
             { 'target' => t,
               'hook_proc' => hook.call(opts.merge(options), t, self) }
+          # Catch and immediately re-raise noop errors so the entire plan fails,
+          # otherwise we'll get a failing result for each target. This error is
+          # raised by the plugin module, since that's where the plugin task is
+          # loaded and where Bolt checks if it can be run in noop mode.
+          rescue Bolt::Plugin::Module::NoopError
```

Comment:
Since this can also come from the Task plugin this probably needs to go in the Plugin class

Reply:
Also sorry I tried to leave these as comments and Github wanted it to be a review :P